### ISSUE TITLE
Check argument names

### ISSUE THREAD

**Issue by cushon** [google/error-prone#128]
_[Original issue](https://code.google.com/p/error-prone/issues/detail?id=128) created by **impjdi@google.com** on 2013-04-29 at 12:11 AM_

---

TL;DR: If same types and pretty similar argument names which are swapped, throw some warning.

Sorry if this a duplicate request... I searched with the keyword arguments and didn't find any matches (or overlooked).

This might be tricky (in terms of setting the threshold for name similarities) and cause false positives, but maybe a useful test.

class A {
&nbsp;&nbsp;public A(int ramSize, int diskSize) {
&nbsp;&nbsp;&nbsp;&nbsp;// do something
&nbsp;&nbsp;}
}

class B extends A {
&nbsp;&nbsp;public B(int ramSize, int diskSize) {
&nbsp;&nbsp;&nbsp;&nbsp;super(diskSize, ramSize);  // swapped args for whatever reason
&nbsp;&nbsp;&nbsp;&nbsp;// do something else
&nbsp;&nbsp;}
}

or 

class B extends A {
&nbsp;&nbsp;public B(int diskSize, int ramSize) {
&nbsp;&nbsp;&nbsp;&nbsp;super(diskSize, ramSize);  // swapped args because author confused with his own class' args order
&nbsp;&nbsp;&nbsp;&nbsp;// do something else
&nbsp;&nbsp;}
}

It doesn't have to be the constructor, but any sort of method calls.

**Comment by cushon** [comment]
We have this for constructors: http://errorprone.info/bugpattern/ChainingConstructorIgnoresParameter

It might be hard to generalize to other methods with a low false positive rate.

**Comment by graememorgan** [comment]
This exists under the argumentselectiondefects package.
